skimage/metrics/overlap.py

```diff
@@ -0,0 +1,251 @@
+
+    bottom_right : array-like of ints or floats, optional
+        (r,c)-coordinates for the bottom right corner of the BoundingBox.
+
+    dimensions : array-like of ints or floats, optional
+        dimensions of the BoundingBox (height, width). The default is None.
+
+    Raises
+    ------
+    ValueError
+        If neither or both of bottom_right and dimensions are provided.
+
+    Attributes
+    ----------
+    top_left : array of int or float
+        The top left corner of the BoundingBox.
+    bottom_right : array of int or float
+        The bottom right corner of the BoundingBox.
+    """
+
+    def __init__(self, top_left, *, bottom_right=None, dimensions=None):
+        self.top_left = np.asarray(top_left)
+
+        if (bottom_right is None) and (dimensions is None):
+            raise ValueError("Specify one of bottom_right or dimensions.")
+
+        if (bottom_right is not None) and (dimensions is not None):
+            raise ValueError("Specify bottom_right or dimensions, not both.")
+
+        if bottom_right is not None:
+            if not np.all(top_left <= bottom_right):
+                raise ValueError("Bottom right corner should have coordinates "
+                                 "larger or equal to the top left corner.")
+            self.bottom_right = np.asarray(bottom_right)
+
+        elif dimensions is not None:
+            dimensions = np.asarray(dimensions)
+            if not (dimensions >= 0).all():
+                raise ValueError("Dimensions should be positive.")
+            self.bottom_right = self.top_left + dimensions
+
+    @property
+    def height(self):
+        # use negative indexing in anticipation of nD hyperBoundingBoxes.
+        return self.bottom_right[-2] - self.top_left[-2]
+
+    @property
+    def width(self):
+        # use negative indexing in anticipation of nD hyperBoundingBoxes.
+        return self.bottom_right[-1] - self.top_left[-1]
+
+    @property
+    def ndim(self):
+        return len(self.top_left)
+
+    def __bool__(self):
+        return True
+
+    def __eq__(self, other: BoundingBox):
+        """Return true if 2 BoundingBoxes have the same position and dimension."""
+        if not isinstance(other, BoundingBox):
+            raise TypeError(
+                    'Equality can only be checked with another BoundingBox'
+                    )
+
+        if self.ndim != other.ndim:
+            return False
+
+        return (np.all(self.top_left == other.top_left)
+                and np.all(self.bottom_right == other.bottom_right))
+
+    def __repr__(self):
+        return (f'BoundingBox({tuple(self.top_left)}, '
+                f'bottom_right={tuple(self.bottom_right)})')
+
+    def __str__(self):
+        return self.__repr__()
+
+    @property
+    def area(self):
+        """Return the area of a 2D BoundingBox."""
+        if self.ndim == 2:
+            return self.integral
+
+        raise NotImplementedError("Area is only defined for 2D.")
```

Comment:
Why not provide the area for a higher dimensional BBox as well?

Reply:
Thanks for the feedback. You mean the area for a higher dim BBox would be the sum of all the surfaces areas ?
e.g. a 3D bounding box we would return the sum of the area for the 6 faces of the box ?
That would make sense, if we document it accordingly.

We briefly discuss this terminology issue below, see @jni  comment there 

> We have discussed this problem with terminology not being properly n-dimensional before [here](https://github.com/scikit-image/scikit-image/issues/3812) in the context of regionprops. My suggestion then would be to do the same here: we have a single object called a Box (though I'm still ok with a Rectangle), and it has .area, .ndim, and .volume, and the area is not defined above 2D. I would argue that .volume should evaluate to .area in 2D because then that one attribute works for nD. (afaik the 'volume' of a 4D hyperrectangle is the product of the extents across all dimensions, not just 3

Currently, area is only defined for 2D and volume for 3D.